### PR TITLE
BaseSerializer Implemented

### LINKED ISSUE #12: Implement a base serializer class for API views
Django-rest-framework, which has been used to create API views, does not call the `full_clean` method under models, this means that custom model validations would not run on API views while creating or updating records. We need to create a base serializer class which can be extended to other apps.

### PR DESCRIPTION
**Pull request by atb00ker**
Added new file `openwisp_utils/api/serializers.py` and implemented a base serializer that can be used in other projects.

Fixes #12

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -33,7 +33,7 @@
     packages=find_packages(exclude=['tests', 'docs']),
     include_package_data=True,
     zip_safe=False,
-    install_requires=['django-model-utils'],
+    install_requires=['django-model-utils', 'djangorestframework'],
```
Comment (nemesisdesign) [inline setup.py]: not all the packages using `openwisp-utils` have APIs and adding this here would make them download DRF even when not needed.

I suggest the following solution:

- remove this requirement here
- add djangorestframework to `requirements-test.txt`
- add a try/except `ImportError` block when you import `serializers` in `openwisp_utils/api/serializers.py`, see my other comment

File: openwisp_utils/api/serializers.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+from rest_framework import serializers
```
Comment (nemesisdesign) [inline openwisp_utils/api/serializers.py]: following the text of the other comment, here you could do something like:

```python
try:
    from rest_framework import serializers
except ImportError:  # pragma: nocover
    import sys
    print('Django REST Framework is required to use this feature but it is not installed')
    sys.exit(1)
```

Comment (atb00ker) [reply openwisp_utils/api/serializers.py]: How about using django's abstraction for this?

```
from django.core.exceptions import ImproperlyConfigured

try:
    from rest_framework import serializers
except ImportError:  # pragma: nocover
    raise ImproperlyConfigured('Django REST Framework is required to use this feature but it is not installed')
```

Comment (nemesisdesign) [reply openwisp_utils/api/serializers.py]: sure, a lot better! :+1:

**Comment by atb00ker** [comment]
Made the required changes. :smile:

File: openwisp_utils/api/serializers.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+from django.core.exceptions import ImproperlyConfigured
+
+try:
+    from rest_framework import serializers
+except ImportError:  # pragma: nocover
+    raise ImproperlyConfigured(
+        'Django REST Framework is required to use this feature but it is not installed')
```
Comment (nemesisdesign) [inline openwisp_utils/api/serializers.py]: we usually break this kind of code into multiple lines with something like:

```python
raise ImproperlyConfigured('Django REST Framework is required to use '
                           'this feature but it is not installed')
```

Could you update these 2 lines for consistency reasons please?

**Pull request merged**